formats/config/src/main/kotlin/org/jetbrains/kotlinx/serialization/config/ConfigReader.kt

```diff
@@ -76,10 +79,14 @@ class ConfigParser(context: SerialModule = EmptyModule): AbstractSerialFormat(co
         private fun composeName(parentName: String, childName: String) =
             if (parentName.isEmpty()) childName else parentName + "." + childName
 
-        override fun SerialDescriptor.getTag(index: Int): String = composeName(
-            currentTagOrNull
-                    ?: "", getElementName(index)
-        )
+        override fun SerialDescriptor.getTag(index: Int): String =
+            composeName(currentTagOrNull ?: "", getConventionElementName(index))
+
+        private fun SerialDescriptor.getConventionElementName(index: Int): String {
+            val originalName = getElementName(index)
+            return if (!configuration.useConfigNamingConvention || getElementAnnotations(index).any { it is SerialName }) originalName
```

Comment:
unfortunately, `SerialName` is not stored in element annotations, since it is not a `@SerialInfo` annotation. Instead, it affects descriptor directly — `getElementName(index)` returns value from `@SerialName` annotation, if present. I'm afraid there's currently no way to check whether serial name was altered, since it was made indistinguishable from property name on purpose. Probably, test regex for match would be sufficient for your use case?

Reply:
Yeah, it would be enough. I thought, that it would be nice to use overall the naming convention and still could overwrite the `SerialName`. Maybe this can be achieved in the future with a `@ConfigSerialName` annotation.

Reply:
In my XML format I also have the function that I need to know whether SerialName was overridden or not. If there is a default way this is done/visible that would be good.

ps. as context, if the value to be serialized is an xml tag, it will normally want to use the name derived from the type that stores the value (analogous with how XML Schema/DTDs work), not from the attribute that contains it. However if the attribute was renamed, then by default the assumption would be that the user intended for that name to be used.